src/fortranscanner.l

```diff
@@ -672,11 +675,9 @@ extends{ARGS}                           {
                                         }
 public                                  {
                                           yyextra->current->protection = Public;
-                                          yyextra->typeProtection = Public;
                                         }
```

Comment:
Why is the type protection removed here?

Reply:
This only refers to the visibility of the type itself (thats the line `type, public :: T1`) and does not change the default for the member variables of the type... I've added a case in the test to check that.

Reply:
(but I'll have a look at the standard again)

Reply:
Is a bit against my feeling.
gfortran can have a bug but it is a good indication.
- Which version of gfortran did you use to test?

Reply:
Now that you asked: same behavior for gfortran 9.4, 12.1 and current ifort.

Reply:
Thanks, gives some more confidence.